File: backend/src/main/java/com/joying/chat/dto/ChatRoomStatusEvent.java
package com.joying.chat.dto;

import java.time.Instant;

import com.joying.chat.domain.ChatRoomStatus;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 채팅방 상태가 바뀐 것을 상대방에게 알리는 이벤트.
 *
 * <p>나가기, 재입장, 자동 종료가 여기로 나간다.
 */
@Getter
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
public class ChatRoomStatusEvent {

	private Long chatRoomId;
	private EventType eventType;

	/** 이벤트를 발생시킨 사용자 */
	private Long memberId;

	private String memberNickname;

	/** 자동 종료처럼 방 상태가 함께 바뀌는 경우에만 채운다 */
	private ChatRoomStatus status;

	@Builder.Default
	private Instant timestamp = Instant.now();

	public enum EventType {
		/** 상대방이 나갔다 */
		MEMBER_LEFT,

		/** 상대방이 다시 들어왔다 */
		MEMBER_REJOINED,

		/** 채팅방이 자동으로 종료됐다 */
		ROOM_CLOSED
	}
}

File: backend/src/main/kotlin/com/joying/chat/dto/ChatRoomStatusEvent.kt
package com.joying.chat.dto

import com.joying.chat.domain.ChatRoomStatus
import java.time.Instant

/**
 * 채팅방 상태 변경 실시간 알림 이벤트
 *
 * 채팅방 나가기, 재입장, 자동 종료 등의 이벤트를 상대방에게 실시간 알림
 */
data class ChatRoomStatusEvent(
    val chatRoomId: Long,
    val eventType: EventType,
    val memberId: Long, // 이벤트를 발생시킨 사용자 ID
    val memberNickname: String?, // 이벤트를 발생시킨 사용자 닉네임 (선택)
    val status: ChatRoomStatus? = null, // 채팅방 상태 (자동 종료 시)
    val timestamp: Instant = Instant.now(),
) {
    enum class EventType {
        MEMBER_LEFT, // 상대방이 나갔습니다
        MEMBER_REJOINED, // 상대방이 다시 들어왔습니다
        ROOM_CLOSED, // 채팅방이 자동 종료되었습니다
    }
}
